### PR TITLE
feat: add scheduler related metrics

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds a scheduler metric and enables the queue metrics for the scheduler workqueue.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- [x] Unit tests

### Special notes for your reviewer

N/A

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -50,3 +50,22 @@ var (
...
+// The scheduler related metrics.
+var (
+	// SchedulingCycleDurationMilliseconds is a Fleet scheduler metric that tracks how long it
+	// takes to complete a scheduling loop run.
+	SchedulingCycleDurationMilliseconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "scheduling_cycle_duration_milliseconds",
+			Help: "The duration of a scheduling cycle run in milliseconds",
+			Buckets: []float64{
+				10, 50, 100, 500, 1000, 5000, 10000, 50000,
```
Comment (ryanzhang-oss) [inline pkg/metrics/metrics.go:63]: any reason to use the this bucket size?

**Review by ryanzhang-oss — APPROVED** [review]

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (michaelawyu) [reply pkg/metrics/metrics.go:63]: Oh, the unit is milliseconds and based on prev. observations these buckets seem to be appropriate => would you prefer finer bucket sizes?

Comment (ryanzhang-oss) [reply pkg/metrics/metrics.go:63]: how about copy controller-runtime one?

Comment (michaelawyu) [reply pkg/metrics/metrics.go:63]: Oh, the controller-runtime one uses custom ExponetialBuckets (`prometheus.ExponentialBuckets(10e-9, 10, 10)`) and the unit is in seconds -> I can send a separate PR if you'd like to switch to seconds as well

**Comment by michaelawyu** [comment]
I'll merge it first to unblock the progress; if there's any concern and further requested change, please let me know.

**Pull request merged**